tools/torch/main.lua

```diff
@@ -313,50 +379,20 @@ if lfs.mkdir(paths.concat(opt.save)) then
     logmessage.display(0,'created a directory ' .. paths.concat(opt.save) .. ' to save all the snapshots')
 end
 
-logmessage.display(0,'creating worker threads')
--- create data loader for training dataset
-trainDataLoader = DataLoader:new(4, -- num threads
-                                 package.path,
-                                 opt.dbbackend, opt.train, opt.train_labels,
-                                 opt.mirror, opt.crop,
-                                 opt.croplen, meanTensor,
-                                 true, -- train
-                                 opt.shuffle,
-                                 classes ~= nil -- whether this is a classification task
-                                 )
-
--- retrieve info from train DB
-local trainSize, imageSizeX, imageSizeY = trainDataLoader:getInfo()
-
-logmessage.display(0,'found ' .. trainSize .. ' images in train db' .. opt.train)
-
-local validationDataLoader, valSize
-
-if opt.validation ~= '' then
-    local imageSizeX, imageSizeY
-    validationDataLoader = DataLoader:new(4, -- num threads
-                                          package.path,
-                                          opt.dbbackend, opt.validation, opt.validation_labels,
-                                          false, -- no need to do random mirrorring
-                                          opt.crop, opt.croplen,
-                                          meanTensor,
-                                          false, -- train
-                                          false, -- shuffle
-                                          classes ~= nil -- whether this is a classification task
-                                          )
-    valSize, imageSizeX, imageSizeY = validationDataLoader:getInfo()
-    logmessage.display(0,'found ' .. valSize .. ' images in train db' .. opt.validation)
-end
-
 -- validate "crop length" input parameter
-if opt.crop then
-    if opt.croplen > imageSizeY then
-        logmessage.display(2,'invalid crop length! crop length ' .. opt.croplen .. ' is less than image width ' .. imageSizeY)
+if opt.crop and inputTensorShape then
+    if opt.croplen > inputTensorShape[3] then
+        logmessage.display(2,'invalid crop length! crop length ' .. opt.croplen .. ' is less than image width ' .. inputTensorShape[3])
```

Comment:
No need to throw an error here. You can just make `croplen = min(croplen, width, height)`. That seems to be what Caffe does.